internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go

```diff
@@ -160,14 +185,17 @@ func (s *Server) ApplyConfiguration(ctx context.Context, in *machine.ApplyConfig
 		return nil, err
 	}
 
-	switch {
-	// --immediate
-	case in.Immediate:
+	//nolint:exhaustive
+	switch in.Mode {
+	// --mode=no-reboot
+	case machine.ApplyConfigurationRequest_NO_REBOOT:
 		if err := s.Controller.Runtime().SetConfig(cfgProvider); err != nil {
 			return nil, err
 		}
-	// default, no `--on-reboot`
-	case !in.OnReboot:
+	// --mode=staged
+	case machine.ApplyConfigurationRequest_STAGED:
+	// --mode=reboot
+	default:
```

Comment:
should this be explicit `--mode=reboot` vs. `default:` ? 

and other modes should not be available at this point (so we could through an error)